src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -1842,10 +1825,10 @@ because xbuild doesn't support framework reference assemblies.
   <CopyIfChanged
     SourceFiles="@(_ResolvedAssemblies->'%(Identity).config')"
     DestinationFiles="@(_ShrunkAssemblies->'%(Identity).config')" />
-  
+
   <!-- Shrink Mono.Android.dll by removing attribute only needed for GenerateJavaStubs -->
   <RemoveRegisterAttribute
-    Condition="'$(AndroidLinkMode)' != 'None' AND '$(AndroidUseSharedRuntime)' != 'true'"
+    Condition="'$(AndroidLinkMode)' != 'None' AND '$(Optimize)' == 'true'"
```

Comment:
I think we should use `$(AndroidIncludeDebugSymbols)` here instead of `$(Optimize)`.